File: law-service/src/main/java/avg/vnlaw/lawservice/entities/Chapter.java
package avg.vnlaw.lawservice.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table( name ="pdchapter")
@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@Builder
public class Chapter extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(nullable = false)
    private String name;

    @Column(name = "`index`", nullable = false)
    private String index;

    @Column(name = "`order`", nullable = false)
    private Integer order;

    @JsonIgnoreProperties({"hibernateLazyInitializer","handler"})
    @ManyToOne(fetch =  FetchType.LAZY)
    @JoinColumn(name = "subject_id", nullable = false)
    private Subject subject;
}

File: front-end/mobile/lib/data/models/vbqppl/chapter.dart
class Chapter {
  final String id;
  final String name;
  final String index;
  final int order;

  Chapter({
    required this.id,
    required this.name,
    required this.index,
    required this.order
  });

  factory Chapter.fromJson(Map<String, dynamic> json) {
    return Chapter(
      id: json['id'],
      name: json['name'],
      index: json['index'] ?? '',
      order: json['order'],
    );
  }
}
